staging/src/k8s.io/client-go/tools/events/helper.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package events
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	eventsv1 "k8s.io/api/events/v1"
+	eventsv1beta1 "k8s.io/api/events/v1beta1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// GetFieldSelector returns the appropriate field selector based on the API version being used to communicate with the server.
+// The returned field selector can be used with List and Watch to filter desired events.
+func GetFieldSelector(groupVersion schema.GroupVersion, regardingName, regardingNamespace, regardingKind, regardingUID string) (fields.Selector, error) {
+	field := fields.Set{}
+	mapping := map[schema.GroupVersion]string{
+		eventsv1.SchemeGroupVersion:      "regarding",
+		eventsv1beta1.SchemeGroupVersion: "regarding",
+		corev1.SchemeGroupVersion:        "involvedObject",
+	}
+
+	if _, ok := mapping[groupVersion]; !ok {
+		return field.AsSelector(), fmt.Errorf("unknown version %v", groupVersion)
+	}
+
+	prefix := mapping[groupVersion]
+	if len(regardingName) > 0 {
+		field[fmt.Sprint(prefix, ".name")] = regardingName
```

Comment:
`prefix + ".name"` is simpler/faster (same applies for other `Sprint` calls)

Reply:
Thank you very much, I have modified these